tests/test_utils.py

```diff
@@ -135,6 +135,40 @@ def test_decomposition(self, hamiltonian):
         assert np.allclose(hamiltonian, linear_comb)
 
 
+class TestSparse:
+    """Tests the sparse_hamiltonian function"""
+
+    def test_sparse_typeerror(self):
+        """Tests that sparse_hamiltonian raises an error if the given Hamiltonian is not of type
+        `qml.Hamiltonian`"""
+
+        with pytest.raises(TypeError, match="Passed Hamiltonian must be of type"):
+            pu.sparse_hamiltonian(np.eye(2))
```

Comment:
Wait, what is `pu` here? We're importing `pennylane.utils` as `pu`? Weird

Reply:
It was already there. I will use `pennylane.utils`.